internal/gitinterface/tree.go

```diff
@@ -197,3 +278,130 @@ func (t *TreeBuilder) writeTrees(parent string, tree *object.Tree) (plumbing.Has
 
 	return WriteTree(t.r, tree.Entries)
 }
+
+// ReplacementTreeBuilder is used to create multi-level trees in a repository.
+// Based on `buildTreeHelper` in go-git.
+// This will replace TreeBuilder above, and will be renamed at that point.
+type ReplacementTreeBuilder struct {
```

Comment:
bikeshed: How does this different from the existing TreeBuilder? If it's the multi-levels, would `MultilevelTreeBuilder` be a better name?

Reply:
This replaces TreeBuilder; it's largely the same but doesn't use go-git. The plan is:
a) add git binary based APIs to the package
b) update other packages to use the new API (#372)
c) remove old, now unused APIs, also rename some things (ReplacementTreeBuilder -> TreeBuilder, TestRepositoryCommit -> TestCommit, etc.)